### PR TITLE
Add abseil-cpp github dependency (for: flag-parsing in C API based cjxl)

### PR DESCRIPTION
**Pull request by fischbacher**

### CODE REVIEW

**Review by fischbacher — COMMENTED** [review]
Looks as expected.

File: third_party/CMakeLists.txt [hunk 1]
```diff
@@ -111,6 +111,17 @@ include(lodepng.cmake)
 configure_file("${CMAKE_CURRENT_SOURCE_DIR}/lodepng/LICENSE"
                ${PROJECT_BINARY_DIR}/LICENSE.lodepng COPYONLY)
 
+# abseil-cpp
+if( NOT EXISTS "${CMAKE_CURRENT_SOURCE_DIR}/abseil-cpp/absl/base/options.h" )
+  message(FATAL_ERROR "Please run ${PROJECT_SOURCE_DIR}/deps.sh to fetch the "
+          "build dependencies.")
+endif()
+set(ABSL_BUILD_TESTING OFF CACHE BOOL "Build tests" FORCE)
```
Comment (veluca93) [inline third_party/CMakeLists.txt:119]: Why FORCE?

**Comment by veluca93** [comment]
We should probably add an equivalent of LIBJXL_FORCE_SYSTEM_BROTLI also for absl, and probably modify the debian/ files and/or the CI configuration accordingly.

File: third_party/CMakeLists.txt (hunk 1 above)
Comment (fischbacher) [reply third_party/CMakeLists.txt]: Should not be needed, removed.

**Comment by veluca93** [comment]
Can we wait to submit this until we have support to system ABSL? (i.e. LIBJXL_FORCE_SYSTEM_ABSL)

**Comment by mo271** [comment]
Seems like this is enough for using absl (tried with @fischbacher further changes for the new encoder)

**Comment by data-man** [comment]
So big library for flag-parsing only?
https://github.com/CLIUtils/CLI11 seems much lighter.

**Comment by vtorri** [comment]
@data-man i agree with you, and even more : parsing options of a program with str(n)cmp, or with std::string::compare() method, is just trivial.

**Comment by mo271** [comment]
We decided to go with gflags instead. Thanks @data-man and @vtorri for participating by commenting.

**Pull request closed without merging**